jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt

```diff
@@ -77,25 +77,36 @@ class LocalLambdaRunConfiguration(project: Project, factory: ConfigurationFactor
     }
 
     override fun checkConfiguration() {
-        checkSamVersion()
         resolveRegion()
         resolveCredentials()
-        checkLambdaHandler()
+        val (runtime) = checkLambdaHandler()
+        checkSamVersion(runtime)
         checkInput()
     }
 
-    private fun checkSamVersion() {
-        ExecutableManager.getInstance().getExecutableIfPresent<SamExecutable>().let {
+    private fun checkSamVersion(runtime: Runtime) {
+        val executable = ExecutableManager.getInstance().getExecutableIfPresent<SamExecutable>().let {
             when (it) {
                 is ExecutableInstance.Executable -> it
                 is ExecutableInstance.InvalidExecutable, is ExecutableInstance.UnresolvedExecutable -> throw RuntimeConfigurationError(
                     (it as? ExecutableInstance.BadExecutable)?.validationError
                 )
             }
         }
+
+        runtime.runtimeGroup?.let { runtimeGroup ->
+            SemVer.parseFromText(executable.version)?.let { semVer ->
+                // TODO: Executable manager should better expose the VersionScheme of the Executable...
+                try {
+                    runtimeGroup.validateSamVersion(runtime, semVer)
+                } catch (e: Exception) {
+                    throw RuntimeConfigurationError(e.message)
+                }
+            }
+        }
     }
 
-    private fun checkLambdaHandler() {
+    private fun checkLambdaHandler(): Pair<Runtime, String> {
```

Comment:
Do we use the second return anywhere? what is it for?

Reply:
Nah, I can remove it